spec/operators/bufferToggle-spec.ts

```diff
@@ -341,4 +342,54 @@ describe('Observable.prototype.bufferToggle', () => {
     expectSubscriptions(e1.subscriptions).toBe(e1subs);
     expectSubscriptions(e2.subscriptions).toBe(e2subs);
   });
+
+  it('should accept closing selector returns promise resolves', (done: DoneSignature) => {
+    const e1 = Observable.concat(Observable.of(1),
+      Observable.timer(10).mapTo(2),
+      Observable.timer(10).mapTo(3),
+      Observable.timer(100).mapTo(4)
+      );
+    const expected = [[1]];
+
+    e1.bufferToggle(Observable.of(10), () => new Promise((resolve: any) => { resolve(42); }))
+      .subscribe((x) => {
+        expect(x).toEqual(expected.shift()); },
+        done.fail,
+        () => {
+          expect(expected.length).toBe(0);
+          done();
+      });
+  });
+
+  it('should accept closing selector returns promise rejects', (done: DoneSignature) => {
```

Comment:
I would add the word `that` here. "should accept closing selector **that** returns **a resolved** promise"